Code/Framework/AzCore/AzCore/DOM/DomValue.cpp

```diff
@@ -1177,4 +1178,124 @@ namespace AZ::Dom
     {
         return m_value;
     }
+
+    Value& Value::operator[](const PathEntry& entry)
+    {
+        if (entry.IsEndOfArray())
+        {
+            Array::ContainerType& array = GetArrayInternal();
+            array.push_back();
+            return array[array.size() - 1];
+        }
+        return entry.IsIndex() ? operator[](entry.GetIndex()) : operator[](entry.GetKey());
+    }
+
+    const Value& Value::operator[](const PathEntry& entry) const
+    {
+        return entry.IsIndex() ? operator[](entry.GetIndex()) : operator[](entry.GetKey());
+    }
+
+    Value& Value::operator[](const Path& path)
+    {
+        Value* value = this;
+        for (const PathEntry& entry : path)
+        {
+            value = &value->operator[](entry);
+        }
+        return *value;
+    }
+
+    const Value& Value::operator[](const Path& path) const
+    {
+        const Value* value = this;
+        for (const PathEntry& entry : path)
+        {
+            value = &value->operator[](entry);
```

Comment:
`value` is a pointer type so it would be more &(*value[entry]) - I don't have a strong preference tbh, I just tend to avoid ref and deref at the same time

Reply:
Wouldn't it then just be `value = value[entry];`?

Reply:
It would be `value = (*value)[entry];`